mParticle-Primer/MPKitPrimer.m

```diff
@@ -0,0 +1,181 @@
+    
+    _configuration = configuration;
+    
+    if (startImmediately) {
+        [self start];
+    }
+    
+    return self;
+}
+
+- (void)start {
+    
+    static dispatch_once_t kitPredicate;
+
+    dispatch_once(&kitPredicate, ^{
+        
+        NSString *token = self.configuration[@"apiKey"];
+        
+        [Primer startWithToken:token];
+
+        _started = YES;
+        
+        BOOL automaticPresentation = [self.configuration[@"autoPresent"] boolValue];
+        if (automaticPresentation) {
+            [Primer presentExperience];
+        }
+
+        dispatch_async(dispatch_get_main_queue(), ^{
+            NSDictionary *userInfo = @{mParticleKitInstanceKey: [[self class] kitCode]}
+            [[NSNotificationCenter defaultCenter] postNotificationName:mParticleKitDidBecomeActiveNotification object:nil userInfo:userInfo];
+        });
+    });
+}
+
+- (id const)providerKitInstance {
+    
+    return nil;
+}
+
+#pragma mark - Application
+
+- (nonnull MPKitExecStatus *)continueUserActivity:(nonnull NSUserActivity *)userActivity restorationHandler:(void(^ _Nonnull)(NSArray * _Nullable restorableObjects))restorationHandler {
+    
+    [Primer continueUserActivity:userActivity];
+    
+    return [self statusWithCode:MPKitReturnCodeSuccess];
+}
+
+#pragma mark - User attributes and identities
+
+- (nonnull MPKitExecStatus *)setUserAttribute:(nonnull NSString *)key value:(nullable NSString *)value {
+    
+    if (!value) {
+        return [self statusWithCode:MPKitReturnCodeFail];
+    }
+    
+    NSString *prefixedKey = [NSString stringWithFormat:@"mParticle.%@", key];
+    [Primer appendUserProperties:@{prefixedKey: value}];
+    
+    return [self statusWithCode:MPKitReturnCodeSuccess];
+}
+
+#pragma mark - e-Commerce
+
+- (nonnull MPKitExecStatus *)logCommerceEvent:(nonnull MPCommerceEvent *)commerceEvent {
+    
+    MPKitExecStatus *status = [[MPKitExecStatus alloc] initWithSDKCode:[[self class] kitCode] returnCode:MPKitReturnCodeSuccess forwardCount:0];
+    
+    NSArray *expandedInstructions = [commerceEvent expandedInstructions];
+    for (MPCommerceEventInstruction *commerceEventInstruction in expandedInstructions) {
+        [self logEvent:commerceEventInstruction.event];
+        [status incrementForwardCount];
+    }
+    
+    return status;
+}
+
+#pragma mark - Events
+
+- (nonnull MPKitExecStatus *)logEvent:(nonnull MPEvent *)event {
+    
+    NSMutableDictionary *parameters = [NSMutableDictionary dictionaryWithObject:@"mParticle" forKey:@"pmr_event_api"];
+    
+    if (event.info) {
+        [parameters addEntriesFromDictionary:event.info];
+    }
+    
+    [Primer trackEventWithName:event.name parameters:parameters];
+    
+    return [self statusWithCode:MPKitReturnCodeSuccess];
+}
+
+- (nonnull MPKitExecStatus *)logScreen:(nonnull MPEvent *)event {
+    
+    MPKitExecStatus *status = [[MPKitExecStatus alloc] initWithSDKCode:[[self class] kitCode] returnCode:MPKitReturnCodeSuccess forwardCount:0];
```

Comment:
In this case `logEvent:` will return an instance of `MPKitExecStatus`. If the forward count gets incremented, it will report 2 operations rather than one. My suggestion would be to implement as:
```
MPKitExecStatus *status = [self logEvent:event];
```